BParkingNano/plugins/SimpleFlatTableProducer.cc

```diff
@@ -0,0 +1,246 @@
+                }
+            protected:
+                edm::EDGetTokenT<edm::ValueMap<TIn>> token_;
+        };
+        typedef ValueMapVariable<int> IntExtVar;
+        typedef ValueMapVariable<float> FloatExtVar;
+        typedef ValueMapVariable<double,float> DoubleExtVar;
+        typedef ValueMapVariable<bool,uint8_t> BoolExtVar;
+        typedef ValueMapVariable<int,uint8_t> UInt8ExtVar;
+        boost::ptr_vector<ExtVariable> extvars_;
+
+};
+
+template<typename T>
+class EventSingletonSimpleFlatTableProducer : public SimpleFlatTableProducerBase<T,T> {
+    public:
+        EventSingletonSimpleFlatTableProducer( edm::ParameterSet const & params ):
+            SimpleFlatTableProducerBase<T,T>(params) {}
+
+        ~EventSingletonSimpleFlatTableProducer() override {}
+
+        std::unique_ptr<nanoaod::FlatTable> fillTable(const edm::Event &, const edm::Handle<T> & prod) const override {
+            auto out = std::make_unique<nanoaod::FlatTable>(1, this->name_, true, this->extension_);
+            std::vector<const T *> selobjs(1, prod.product());
+            for (const auto & var : this->vars_) var.fill(selobjs, *out);
+            return out;
+        }
+};
+
+template<typename T>
+class FirstObjectSimpleFlatTableProducer : public SimpleFlatTableProducerBase<T, edm::View<T>> {
+    public:
+        FirstObjectSimpleFlatTableProducer( edm::ParameterSet const & params ):
+          SimpleFlatTableProducerBase<T, edm::View<T>>(params) {}
+
+        ~FirstObjectSimpleFlatTableProducer() override {}
+
+        std::unique_ptr<nanoaod::FlatTable> fillTable(const edm::Event &iEvent, const edm::Handle<edm::View<T>> & prod) const override {
+            auto out = std::make_unique<nanoaod::FlatTable>(1, this->name_, true, this->extension_);
+            std::vector<const T *> selobjs(1, & (*prod)[0]);
+            for (const auto & var : this->vars_) var.fill(selobjs, *out);
+            return out;
+        }
+};
+
+#include "DataFormats/Candidate/interface/Candidate.h"
+typedef SimpleFlatTableProducer<reco::Candidate> SimpleCandidateFlatTableProducer;
+
+#include "SimDataFormats/GeneratorProducts/interface/GenEventInfoProduct.h"
+typedef EventSingletonSimpleFlatTableProducer<GenEventInfoProduct> SimpleGenEventFlatTableProducer;
+
+#include "SimDataFormats/HTXS/interface/HiggsTemplateCrossSections.h"
+typedef EventSingletonSimpleFlatTableProducer<HTXS::HiggsClassification> SimpleHTXSFlatTableProducer;
+
+#include "DataFormats/Candidate/interface/CompositeCandidate.h"
+typedef SimpleFlatTableProducer<reco::CompositeCandidate> SimpleCompositeCandidateFlatTableProducer;
+
+//not really useful in the end because lowptgsf tracks come with BDT taht is not part of GsfTracks
+#include "DataFormats/GsfTrackReco/interface/GsfTrack.h"
+typedef SimpleFlatTableProducer<reco::GsfTrack> SimpleGsfTrackFlatTableProducer;
+
+#include "FWCore/Framework/interface/MakerMacros.h"
+DEFINE_FWK_MODULE(SimpleCandidateFlatTableProducer);
```

Comment:
This will cause clashes with the official Nano ones, should define only the needed ones

Reply:
removed everything